### PR TITLE
ci: Refactor `dex` & `net` selection

### PR DESCRIPTION
**Pull request by KirilMihaylov**

### CODE REVIEW

**Comment by KirilMihaylov** [comment]
Build counter-part: https://github.com/nolus-protocol/nolus-money-market/actions/runs/6612689457

**Review by gmanev7 — APPROVED** [review]
LGTM

**Comment by KirilMihaylov** [comment]
Full CI run: https://github.com/nolus-protocol/nolus-money-market/actions/runs/6627251932

File: scripts/build-and-optimize.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
```
Comment (gmanev7) [inline scripts/build-and-optimize.sh:2]: Let's make sure the script bails out on the first error. Please set the flag.

File: scripts/build-and-optimize.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    cargo build --release --lib --locked --target wasm32-unknown-unknown --target-dir "/target/"
+done
+
+if test $? -eq 0
+then
+    rm -rf "/artifacts/"*
+
+    for wasm in "/target/wasm32-unknown-unknown/release/"*".wasm"
+    do
```
Comment (gmanev7) [inline scripts/build-and-optimize.sh:21]: why don't we merge the two loops? we know each contract package produces a `<package>.wasm`.

Comment (KirilMihaylov) [reply scripts/build-and-optimize.sh:21]: That is a fair point. It totally flew over my mind.

File: scripts/build-and-optimize.sh (hunk 1 above)
Comment (KirilMihaylov) [reply scripts/build-and-optimize.sh:2]: I agree. Didn't take into account that it won't bail on errors in between.

**Comment by KirilMihaylov** [comment]
Full CI run: https://github.com/nolus-protocol/nolus-money-market/actions/runs/6632011114

**Comment by KirilMihaylov** [comment]
Full CI build: https://github.com/nolus-protocol/nolus-money-market/actions/runs/6638062074

File: scripts/build-and-optimize.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+RUSTFLAGS="-C link-arg=-s ${RUSTFLAGS}"
+
+rm -rf "/target/"
+
+mkdir "/target/"
+
+rm -rf "/artifacts/"*
+
+for contract in $(echo "/code/contracts/"* | sed 's/ /\n/g' | sort)
```
Comment (gmanev7) [inline scripts/build-and-optimize.sh:11]: not sure `$(echo "/code/contracts/"* | sed 's/ /\n/g' | sort)`  would iterate over dirs ... what about `echo` -> `ls`?

Comment (KirilMihaylov) [reply scripts/build-and-optimize.sh:11]: I tested it out, in the shell, through the container and CI. ;) I can of course change it to `ls` too.

**Comment by KirilMihaylov** [comment]
Full CI build: https://github.com/nolus-protocol/nolus-money-market/actions/runs/6638362949

Note: Successful.

File: scripts/build-and-optimize.sh (hunk 3 above)
Comment (gmanev7) [reply scripts/build-and-optimize.sh:11]: me too, the output in the shell gave only
`❯ for c in $(echo "/code/contracts/"* | sed 's/ /\n/g' | sort)
∙ do
∙ echo $c
∙ done
/code/contracts/*
`

**Pull request merged**